mongosql/src/algebrizer/test.rs

```diff
@@ -7454,20 +7454,256 @@ mod limit_or_offset_clause {
 }
 
 mod set_query {
-    use super::{catalog, mir_source_bar, mir_source_foo, AST_QUERY_BAR, AST_QUERY_FOO};
-    use crate::{ast, mir, mir::schema::SchemaCache, usererror::UserError};
+    use super::{
+        catalog, mir_source_bar, mir_source_foo, AST_QUERY_BAR, AST_QUERY_FOO, AST_SOURCE_BAR,
+        AST_SOURCE_FOO,
+    };
+    use crate::schema::{Document, Schema};
+    use crate::{
+        ast, map, mir, mir::schema::SchemaCache, multimap, schema, set,
+        unchecked_unique_linked_hash_map,
+    };
+    use mongosql_datastructures::binding_tuple::DatasourceName;
 
     test_algebrize!(
-        union_distinct_not_allowed,
+        union_distinct_star,
         method = algebrize_set_query,
-        expected = Err(Error::DistinctUnion),
-        expected_error_code = 3006,
+        expected = Ok(mir::Stage::Project(mir::Project {
+            source: Box::new(mir::Stage::Group(mir::Group {
+                source: Box::new(mir::Stage::Set(mir::Set {
+                    operation: mir::SetOperation::UnionAll,
```

Comment:
Ah very interesting revelation here: do we even need `mir::SetOperation::Union`? If we algebrize in such a way that all unions become `UnionAll`s in the `mir`, can we do away with `mir`'s representation of distinct `Union`?